### ISSUE TITLE
Siri Proxy Doesnt Do Anything.

### ISSUE THREAD

**Issue by ryanschulz** [plamoni/SiriProxy#149]
I followed all instructions in the Readme.

When I start SiriProxy it does nothing

Last login: Mon Dec  5 19:05:07 on console
rvRyan-Schulzs-Mac-Pro:~ RyanSchulz$ r
-bash: r: command not found
Ryan-Schulzs-Mac-Pro:~ RyanSchulz$ rvmsudo siriproxy server
Password:
Starting SiriProxy on port 443..

And if I do "Test Siri Proxy" on Siri, she says "Sorry but I do not understand Test Siri Proxy."

And no matter what, the terminal still sticks to "Starting SiriProxy 443"

**Comment by mrmosconi** [comment]
Did you change your iPhone's WiFi settings so the DNS points to the machine where dnsmasq is running?

**Comment by ryanschulz** [comment]
What do you mean? Sorry I have been trying for hours and cannot figure it out. I changed the dnsmasq to redirect guzzoni to 192.168.2.5 (my macs IP) using bbedit, am I suppose to put that as My DNS in my iPhone?

Please don't close until I solve! If it would make it easier I can share my screen to you somehow?

Thanks.

**Comment by mrmosconi** [comment]
Yes. On your iPhone, go to Settings > WiFi, then tap the blue arrow on the right next to your SSID. You should see DNS on that screen. Tap it and change it to the ip of your Mac (192.168.2.5).

**Comment by ryanschulz** [comment]
Alright, so I freshly restarted my Mac, and then changed the DNS on my iPhone to 192.168.2.5

Before I tried anything on the Mac, I tested a command with Siri and It went through. 

So I logged in on my Mac, opened terminal, and typed "rvmsudo siriproxy server"

I entered my password and then it said "starting Siriproxy on port 443" and sticks there. Siri cannot connect now and my iPhone has no Internet access through wifi unless I restart my Mac.

If I close and reopen terminal and retry it, it says port 443 is being use by something else until I restart my mac

**Comment by mrmosconi** [comment]
Well, you can't use the same port from multiple processes, so you need to kill siriproxy and then restart it (you don't have to restart your machine). Since you don't have Internet access after you change the DNS on your iPhone, I suspect that dnsmasq isn't configured correctly. If you have another computer on the same network, change its DNS to point to the Mac running dnsmasq, then try pinging guzzoni.apple.com and make sure it comes back as a local IP.

**Comment by ryanschulz** [comment]
sorry I am new to mac, but i am great a developing for windows. How do you kill siriproxy?

**Comment by mrmosconi** [comment]
Open a terminal and type "ps -ef |grep siri". You should see the process running. On the right it will say "rvmsudo siriproxy server". The second column is the process id (PID). Then type "sudo kill " and whatever the PID is.

**Comment by gotnull** [comment]
I'm getting a strange problem. I've followed all the steps, installed SiriProxy and setup dnsmasq properly as well so I don't think that's the problem.

Is it possible that Apple has figured out a way to stop SiriProxy from working as of today? The problem I'm getting is that I hold the Home button and Siri pops up but immediately stops (as if it's like I've double tapped the Siri microphone).

I can confirm that guzzoni.apple.com is localhost when I change a separate machine's DNS to my Mac IP address.

I've also made sure my DNS is pointing to my Mac IP on my iPhone as well as the certificate being installed correctly.

Also when I run the following from my Mac:

#dig guzzoni.apple.com

I get the following. Can someone confirm this is correct as it's showing my DNS Router IP not my Mac IP next to guzzoni.apple.com

(;; SERVER: 10.1.1.1#53(10.1.1.1) - is that correct or should it say 127.0.0.1?

This is my dnsmasq.conf:

address=/guzzoni.apple.com/10.1.1.5 (my mac ip)

; <<>> DiG 9.6-ESV-R4-P3 <<>> guzzoni.apple.com
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 25460
;; flags: qr rd ra; QUERY: 1, ANSWER: 3, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;guzzoni.apple.com.     IN  A

;; ANSWER SECTION:
guzzoni.apple.com.  29  IN  CNAME   origin.guzzoni-apple.com.akadns.net.
origin.guzzoni-apple.com.akadns.net. 199 IN CNAME reg1.guzzoni-apple.com.akadns.net.
reg1.guzzoni-apple.com.akadns.net. 291 IN A 17.174.4.4

;; Query time: 156 msec
;; SERVER: 10.1.1.1#53(10.1.1.1)
;; WHEN: Sun Dec 11 19:11:45 2011
;; MSG SIZE  rcvd: 119

Another problem, which is a major problem I think is that when I change my iPhone DNS to point to my Mac IP I can't surf the web on my iPhone. Any ideas with this? Could it be the main problem or related?

**Comment by dcinzona** [comment]
get a utility on your phone that can ping. 
Then ping guzzoni.apple.com and see if it points to your siri proxy server or not.

**Comment by gotnull** [comment]
I actually got this working on a fresh installation of Ubuntu. I'm still unable to get it working ony Mac. I'm certain it's a problem with not being able to surf the Internet once I've changed my iPhone DNS to my local machine running SiriProxy.

If I can't surf the web using my Mac's IP via DNS on my iPhone then there's something wrong. I just can't ignite out how to fix my Mac to get around this problem.
